src/Middleware/SpaServices.Extensions/src/StaticFiles/SpaStaticFilesExtensions.cs

```diff
@@ -110,8 +111,8 @@ public static void UseSpaStaticFiles(this IApplicationBuilder applicationBuilder
                 }
             }
 
-            
-            app.UseStaticFiles(staticFileOptions);
+            // We should only serve files when there is no endpoint.
+            app.MapWhen(context => context.GetEndpoint() == null, a => app.UseStaticFiles(staticFileOptions));
```

Comment:
Worlds collide. Shouldn't this check go inside of static files so it doesn't hijack requests that already have routes? Or do we need to start work on static file based endpoints?

Reply:
I'm happy to send another PR that puts this inside of static files if you think that's the right thing to do.

Reply:
> Or do we need to start work on static file based endpoints so you can remove this middleware entirely?

I thought this was that? What part do you think we should make more first class?

Reply:
Yes, please update StaticFiles.

Reply:
Will be in the next iteration.